api/renter.go

```diff
@@ -275,8 +287,33 @@ func (api *API) renterContractsHandler(w http.ResponseWriter, _ *http.Request, _
 
 // renterDownloadsHandler handles the API call to request the download queue.
 func (api *API) renterDownloadsHandler(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+	dlq := api.renter.DownloadQueue()
+
+	// Translate []modules.DownloadInfo to []api.DownloadInfo.
+	downloads := make([]DownloadInfo, len(dlq))
+	for i := range dlq {
+		d := dlq[len(dlq)-i-1]
+
+		// Find the destination of the download.
+		dstwriter, ok := (d.Destination).(*modules.DownloadFileWriter)
+		var dst string
+		if ok {
+			dst = dstwriter.Location
+		} else {
+			dst = "httpresp"
```

Comment:
Typically we like to have a separate bool indicating special use. Instead of setting the dstwriter.Location to a magic value, to get httpresp downloads you should have both an empty dstwriter.Location (which we might prefer to call Destination) and a bool indicating that it's going to be streamed. If the streaming bool is set and the destination is not empty, an error is returned because that's an improper request.